packages/core/src/render3/view_ref.ts

```diff
@@ -30,6 +30,8 @@ interface ChangeDetectorRefInterface extends ChangeDetectorRef {}
 export class ViewRef<T> implements EmbeddedViewRef<T>, ChangeDetectorRefInterface {
   private _appRef: ViewRefTracker|null = null;
   private _attachedToViewContainer = false;
+  /** @internal */
+  _destroyDOMOnAppRefDetach = true;
```

Comment:
```suggestion
  _detachDOMOnAppRefDetach = true;
```
Minor nit: we're not actually destroying any DOM nodes, just removing them from the document. They can still be reattached later.